### DISCUSSION TITLE
Amazon MWS to be deprecated in (far?) future, in favor of Selling Partner API

### DISCUSSION THREAD

**Discussion by ericblade** [ericblade/mws-advanced#182] [General]
https://developer.amazonservices.com

https://github.com/amzn/selling-partner-api-docs

The current MWS is to be deprecated in the near future, and any new development is going into this new API.

I have not examined this in the slightest yet -- just heard about it -- but I feel like it would be proper for me to get in a few of the most desired changes here, and then move on to exploring this new API, and see if there's anything that would need to be written to handle that.  

That said, I would doubt that it would be anything less than years before Amazon is able to turn off MWS, as it says they will eventually.  

A brief look indicates that they have Java and C# code available, but nothing for Javascript at the current time.   If anyone knows of anyone who is working on implementing this new SP API in Javascript / Typescript, we should probably all hook up together.  

Looks like it's time to look to the future!

I've been exploring how to get into the new API, working it out rather slowly, here https://github.com/ericblade/sp-api-simple

**Comment by ericblade** [comment]
FWIW, I've actually put together a pretty surprisingly quick API -- Amazon is providing a pretty great set of stuff to go along with this API, and I'm having some pretty good feelings about it, overall, that I definitely didn't have when I was in need of the MWS API initially (my gut feeling about that was "oh, god, this is a nightmare").  While the new SP-API is largely modeled after the existing MWS API, it seems that someone at Amazon is pretty serious about making sure that this one is pretty easy to use.

I'm not ready to show it off yet, as I haven't tested it in the slightest, so I have no idea what it's shortcomings might be, but I was exploring some solutions to auto-generate as much as possible from the docs, and after only a few hours of poking at it, I've got a complete API, that should automatically update.  About the only thing that I see obvious left to do is figure out a way to mate the TypeScript typings to the API itself.

Of course, once I get into actually using it, I might discover that there are some serious gotchas... or I might be just fine with it.  I'm not sure yet.  The actual process of getting access to the API is quite a bit more involved than it is with the MWS API -- It requires a bunch of mucking around in Amazon AWS settings, and I've been too busy to sit down and set that up.

I have a feeling, though, that I'm going to be quite pleasantly surprised by what i'm tentatively calling "sp-api-simple".  I may not even feel like a "sp-api-advanced" is necessary. :-D

**Comment by DmitryMasley** [comment]
@ericblade  Thanks for your hard work. Really looking forward for results of your investigation!

**Comment by ericblade** [comment]
... of course, the MWS API isn't going to be going away anytime in the particularly near future, but imo i think it's best to figure out what's next quickly.

**Comment by ericblade** [comment]
mws-advanced deprecation notice: https://github.com/ericblade/mws-advanced/issues/195
